scripts/upgrade-test.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+# Runs a test to see what happens if we upgrade consensus
+
+set -euxo pipefail
+export RUST_LOG="${RUST_LOG:-info}"
+
+source ./scripts/setup-tests.sh
+
+server4=$(tail -4 $FM_PID_FILE | head -1)
+server3=$(tail -3 $FM_PID_FILE | head -1)
+server2=$(tail -2 $FM_PID_FILE | head -1)
+server1=$(tail -1 $FM_PID_FILE | head -1)
+
+
+function wait_server_shutdown() {
+  echo "Waiting for $1 to shutdown..."
+  tail --pid=$1 -f /dev/null
+  echo "Server $1 has shutdown."
+}
+
+mine_blocks 110
+await_fedimint_block_sync
+await_all_peers
+
+# test a consensus upgrade
+$FM_MINT_CLIENT signal-upgrade "pass0" $FM_CFG_DIR/server-0/private.salt 0
+$FM_MINT_CLIENT signal-upgrade "pass1" $FM_CFG_DIR/server-1/private.salt 1
+
+mine_blocks 10
+await_fedimint_block_sync
+await_all_peers
```

Comment:
Why do we do this instead of just having `server-2` signal right away? Comment would be nice ...

Reply:
True, I don't need it there.